### ISSUE TITLE
Remove locks from KeyBundle

### ISSUE THREAD

**Issue by jschlyter** [IdentityPython/JWTConnect-Python-CryptoJWT#108]
In production systems it turns out readerwriter locks sometimes ends up trying to release locks multiple times. It would help a lot if we could make `KeyBundle` thread safe without the use of locks.

**Comment by rohe** [comment]
Agree! I don't have any proposal right now. If I come up with something I'll let you know.

**Comment by jschlyter** [comment]
Reading the code for `KeyBundle` there seems to be some wrong assumptions, e.g. that arrays are copied by value and not reference.

**Comment by rohe** [comment]
Agree, should be fixed. You or me ?

**Comment by jschlyter** [comment]
I've started in #110. To remove the locks, we could begin with the read locks.
